src/resources/users/UserForm.tsx

```diff
@@ -26,34 +41,184 @@ const schema = yup.object({
     .optional()
 })
 
-export default function UserForm({ isEdit }: FormProps): React.ReactElement {
-  const valueWithTenYears = DateTime.now()
-    .plus({ years: 10 })
-    .toJSDate()
-    .toISOString()
-  const defaultValues: Omit<
-    _Users,
-    | 'id'
-    | 'createdAt'
-    | 'createdBy'
-    | 'username'
-    | 'lastUpdatedAt'
-    | 'lockoutAttempts'
-  > = {
-    name: '',
-    hashed_password: '',
-    role: 'rco-user',
-    departedDate: valueWithTenYears,
-    is_superuser: false,
-    updateBefore: ''
+type UpdateUserRoleFunction = () => Promise<void>
+interface UserDetails {
+  name: string
+  hashed_password?: string
+  role: UserRole
+  is_superuser: boolean
+  username: string
+  departedDate: string | null
+  lastUpdatedAt: string | null
+  lockoutAttempts: number
+  updateBefore: string
+}
+
+const BASE_URL = process.env.API_BASE_URL_KEY ?? 'http://localhost:8000'
+
+const SaveButtonContext: React.FC<any> = (props: {
+  selectUser: string
+  children: React.ReactNode
+}) => {
+  const [create] = useCreate()
+  const redirect = useRedirect()
+  const createUserRole = async (
+    val: UserDetails,
+    selectUser: string
+  ): Promise<number | undefined> => {
+    const userResponse = await axios.get(
+      `${BASE_URL}/api/tables/_users/rows?_filters=username:${val.username}`
+    )
+    const userId = userResponse.data.data[0]?.id
+    if (!userId) {
+      return
+    }
+    const roleResponse = await axios.get(
+      `${BASE_URL}/api/tables/_users_roles/rows?_filters=user_id:${userId}`
+    )
+    const roleId = roleResponse.data.data[0]?.id
+    if (!roleId) {
+      return
+    }
+
```

Comment:
I think we need more robust handling for the above block, to handle cases where legacy users don't have roles.  In this case, the roleId will be empty.  When this happens, we `POST` a new `_user_role` object with the `user_id` and the new `role id`.

This is how the admin staff will bring on board other users as adoption of VAL grows.

Reply:
Fixed in #1085